From 4c2d9d4efb947347c32fcbd4b8f6f6b407d9ac40 Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Wed, 14 Jan 2015 12:45:35 +0800
Subject: [PATCH] MGS-304-2 [#1461] Do not leave garbase in outside pointer

this is the enhanced fix for gpu kernel panic issue based on previous:
gpu causes kernel panic when running bonnie++ with suspend/resume.

Date: Jan 14, 2015
Signed-off-by: Xianzhong <b07117@freescale.com>
Acked-by: Jason Liu
---
 drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_os.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_os.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_os.c
index 2dbd73d..2ec98dc 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_os.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_os.c
@@ -1628,10 +1628,6 @@ gckOS_AllocateNonPagedMemory(
 
     mdl->addr = addr;
 
-    /* Return allocated memory. */
-    *Bytes = bytes;
-    *Physical = (gctPHYS_ADDR) mdl;
-
     if (InUserSpace)
     {
         mdlMap = _CreateMdlMap(mdl, _GetProcessID());
@@ -1770,6 +1766,10 @@ gckOS_AllocateNonPagedMemory(
 
     MEMORY_UNLOCK(Os);
 
+    /* Return allocated memory. */
+    *Bytes = bytes;
+    *Physical = (gctPHYS_ADDR) mdl;
+
     /* Success. */
     gcmkFOOTER_ARG("*Bytes=%lu *Physical=0x%X *Logical=0x%X",
                    *Bytes, *Physical, *Logical);
@@ -3682,9 +3682,6 @@ gckOS_AllocatePagedMemoryEx(
     mdl->pagedMem   = 1;
     mdl->contiguous = Flag & gcvALLOC_FLAG_CONTIGUOUS;
 
-    /* Return physical address. */
-    *Physical = (gctPHYS_ADDR) mdl;
-
     if (Gid != gcvNULL)
     {
         *Gid = mdl->gid;
@@ -3712,6 +3709,9 @@ gckOS_AllocatePagedMemoryEx(
 
     MEMORY_UNLOCK(Os);
 
+    /* Return physical address. */
+    *Physical = (gctPHYS_ADDR) mdl;
+
     /* Success. */
     gcmkFOOTER_ARG("*Physical=0x%X", *Physical);
     return gcvSTATUS_OK;
-- 
1.8.0

